From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lukas Rieger <code@lukasrieger.com>
Date: Wed, 19 Mar 2026 00:00:00 +0100
Subject: [PATCH] rtc: qpnp: Add debug prints for crash diagnosis

Add pr_err trace prints to qpnp_rtc_probe(), qpnp_rtc_set_time(), and
qpnp_rtc_set_alarm() to diagnose boot crash when qpnp-rtc-write is
enabled. Includes dump_stack() in set_time to capture call chain.

Upstream-Status: Inappropriate [temporary debug patch]
Signed-off-by: Lukas Rieger <code@lukasrieger.com>
---
 drivers/rtc/qpnp-rtc.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/rtc/qpnp-rtc.c b/drivers/rtc/qpnp-rtc.c
index 579877306d92..b0ab275cc5c0 100644
--- a/drivers/rtc/qpnp-rtc.c
+++ b/drivers/rtc/qpnp-rtc.c
@@ -108,6 +108,9 @@ qpnp_rtc_set_time(struct device *dev, struct rtc_time *tm)
 
 	rtc_tm_to_time(tm, &secs);
 
+	pr_err("qpnp-rtc-debug: set_time CALLED, secs=%lu\n", secs);
+	dump_stack();
+
 	value[0] = secs & 0xFF;
 	value[1] = (secs >> 8) & 0xFF;
 	value[2] = (secs >> 16) & 0xFF;
@@ -289,6 +292,8 @@ qpnp_rtc_set_alarm(struct device *dev, struct rtc_wkalrm *alarm)
 	struct qpnp_rtc *rtc_dd = dev_get_drvdata(dev);
 	struct rtc_time rtc_tm;
 
+	pr_err("qpnp-rtc-debug: set_alarm CALLED\n");
+
 	rtc_tm_to_time(&alarm->time, &secs);
 
 	/*
